tests/hostpath_quota_test.go

```diff
@@ -0,0 +1,606 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/container"
+	pvc "github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/persistentvolumeclaim"
+	"github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/pod"
+	"github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/volume"
+	sc "github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/storage/v1/storageclass"
+	"github.com/openebs/dynamic-localpv-provisioner/tests/disk"
+)
+
+const (
+	// DiskImageSize is the default file size(1GB) used while creating backing image
+	DiskImageSize = 1073741824
+)
+
+var _ = Describe("TEST HOSTPATH LOCAL PV", func() {
+	var (
+		pvcObj         *corev1.PersistentVolumeClaim
+		scObj          *storagev1.StorageClass
+		accessModes    = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}
+		capacity       = "2M"
+		xfsHostpathDir = "/var/openebs/integration-test/xfs/"
+		podName        = "busybox-hostpath"
+		label          = "demo=hostpath-pod"
+		pvcName        = "pvc-hp"
+		scNamePrefix   = "sc-hp-xfs"
+		scName         string
+		podObj         *corev1.Pod
+		labelselector  = map[string]string{
+			"demo": "hostpath-pod",
+		}
+	)
+	physicalDisk := disk.Disk{}
+
+	When("preparing the loopback device with xfs fs", func() {
+		It("should create and mount the disk", func() {
+			physicalDisk = PrepareDisk("xfs", xfsHostpathDir)
+		})
+	})
+
+	When("hostpath is not having xfs filesystem", func() {
+		When("pod consuming pvc with a valid quota storageclass is created", func() {
```

Comment:
This way of embedding one When block in another is affecting readability.
You should consider splitting the tests into separate specs, if the test cases are too varied.